### PR TITLE
DRILL-5919: Add NaN/Infinity/-Infinity data types for Json processing

### PR DESCRIPTION
**Pull request by omalashevsky**

### CODE REVIEW

File: framework/resources/Functional/jsonNanInf/negativeCases/queries.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "testId": "jsonNaNInf_negative_cases",
```
Comment (Agirish) [inline framework/resources/Functional/jsonNanInf/negativeCases/queries.json]: We have categories for negative tests (tests which are expected to fail) - so will need to set that if that's your intension.

File: framework/src/main/java/org/apache/drill/test/framework/ColumnList.java [hunk 2]
```diff
@@ -161,12 +161,17 @@ private boolean compare(ColumnList o1, ColumnList o2) {
           }
           break;
         case Types.DOUBLE:
-          double d1 = (Double) list1.get(i);
-          double d2 = (Double) list2.get(i);
-          if ((d1 + d2) / 2 != 0) {
-            if (!(Math.abs((d1 - d2) / ((d1 + d2) / 2)) < 1.0E-12)) return false;
-          } else if (d1 != 0) {
-            return false;
+          Double d1 = (Double) list1.get(i);
```
Comment (Agirish) [inline framework/src/main/java/org/apache/drill/test/framework/ColumnList.java]: For the test framework changes, can you please open a separate PR?

File: framework/resources/Functional/jsonNanInf/negativeCases/queries.json (hunk 1 above)
Comment (Agirish) [reply framework/resources/Functional/jsonNanInf/negativeCases/queries.json]: To clarify, pre-commit runs do not include negative tests. However, we have nightly and release time tests which cover these cases.

Comment (omalashevsky) [reply framework/resources/Functional/jsonNanInf/negativeCases/queries.json]: Thanks for the clarification, the queries.json file has been changed, please verify

File: framework/src/main/java/org/apache/drill/test/framework/ColumnList.java (hunk 2 above)
Comment (omalashevsky) [reply framework/src/main/java/org/apache/drill/test/framework/ColumnList.java]: sure, will do.

Comment (omalashevsky) [reply framework/src/main/java/org/apache/drill/test/framework/ColumnList.java]: The changes have been rolled back, new pull request - https://github.com/mapr/drill-test-framework/pull/420 , could you please review?

**Comment by Agirish** [comment]
+1. Thanks for adding these tests. I'll run this through regression and merge if no issues.

**Comment by prasadns14** [comment]
@Agirish, I think we should plan on a directory structure. Rather than creating multiple higher level folders we could probably move these tests under json folder.

i.e we could have `jsonNanInf` under framework/resources/Functional/json/jsonNanInf/

That way the test cases would be more organized.

**Comment by Agirish** [comment]
Agreed! We should clean-up and what better way than to have new PRs make this change. @omalashevsky can you kindly take a look?

**Comment by omalashevsky** [comment]
@Agirish, the Functional/jsonNanInf folder has been moved to Functional/json/jsonNanInf, could you review? 
Also please pay attention that I didn't touch Datasources/jsonNanInf because I can't see any common folder for json data_sources

**Comment by Agirish** [comment]
Thanks Sasha!

**Comment by Agirish** [comment]
Regression run failed. 
@omalashevsky, we'll need your branch to be rebased on top of latest mapr master. As your changes for processing NaN values just went in.

**Comment by omalashevsky** [comment]
@Agirish , done, could you please review

**Comment by Agirish** [comment]
Test run was successful. Merging.

**Pull request merged**